pkg/runtime/hotreload/reconciler/components.go

```diff
@@ -68,6 +70,14 @@ func (c *components) update(ctx context.Context, comp compapi.Component) {
 	if c.proc.AddPendingComponent(ctx, comp) {
 		log.Infof("Component updated: %s", comp.LogName())
 		c.proc.WaitForEmptyComponentQueue()
+
+		if c.cb != nil {
+			if err := c.cb(registry.ComponentRegistry{
```

Comment:
The callbacks here are at the wrong level- they need to be inside the component processor in `Init` and `Close`. Please return errors.

https://github.com/dapr/dapr/blob/334ae9eea43d487a7b29a0e4aef904e3eba57a10/pkg/runtime/processor/components.go#L30
https://github.com/dapr/dapr/blob/334ae9eea43d487a7b29a0e4aef904e3eba57a10/pkg/runtime/processor/components.go#L51

Reply:
updated